Service/Cart.php

```diff
@@ -242,17 +263,23 @@ public function getAddToCartRequest(string $sku, $qty, $discount = null)
     }
 
     /**
+     * Get determineQty with MSI module
+     *
      * @param $qty
      * @param $product
      * @return float
      */
     protected function determineQty($requestedQty, $product)
     {
+        $salesChannel = $this->saleschannel->create();
+        $getStockItemDatafact = $this->getStockItemDatafact->create();
```

Comment:
@zeel1012 $this->getStockItemDatafact->create(); doesn't return a factory. so you need to correct the variable name.